storm/check.py

```diff
@@ -0,0 +1,688 @@
+            value = v[0]
+            tags = v[1]
+            self.report_gauge(metric=k, value=value, tags=tags, additional_tags=self.additional_tags)
+
+        # Nimbus Stats
+        nimbus_stats = self.get_storm_nimbus_summary()
+        for ns in self.process_nimbus_stats(self.environment_name, nimbus_stats):
+            for k, v in ns.items():
+                value = v[0]
+                tags = v[1]
+                self.report_gauge(metric=k, value=value, tags=tags, additional_tags=self.additional_tags)
+
+        # Supervisor Stats
+        supervisor_stats = self.get_storm_supervisor_summary()
+        for ss in self.process_supervisor_stats(supervisor_stats):
+            for k, v in ss.items():
+                value = v[0]
+                tags = v[1]
+                self.report_gauge(metric=k, value=value, tags=tags, additional_tags=self.additional_tags)
+
+        # Topology Stats
+        summary = self.get_storm_topology_summary()
+        for topology in summary['topologies']:
+            topology_id = topology.get('id')
+            topology_name = _g(topology, 'unknown', None, 'name')
+            if topology_name not in self.excluded_topologies:
+                stats = self.get_topology_info(topology_id=topology_id)
+                interval_results = []
+                for interval in self.intervals:
+                    results = self.process_topology_stats(topology_stats=stats, interval=interval)
+                    if len(results) > 0:
+                        interval_results.append(results)
+                metric_stats = self.get_topology_metrics(topology_id=topology_id)
+                metric_results = self.process_topology_metrics(topology_name, metric_stats)
+
+                topology_status = _g(stats, 'unknown', None, 'status').upper()
+                if topology_status != 'ACTIVE':
+                    check_status = AgentCheck.CRITICAL
+                    self.service_check(
+                        'topology-check.{}'.format(topology_name),
+                        status=check_status,
+                        message='{} topology status marked as: {}'.format(topology_name, topology_status),
+                        tags=['env:{}'.format(self.environment_name),
+                              'environment:{}'.format(self.environment_name)] + self.additional_tags
+                    )
+                else:
+                    check_status = AgentCheck.OK
+                    self.service_check(
+                        'topology-check.{}'.format(topology_name),
+                        status=check_status,
+                        message='{} topology is active'.format(topology_name),
+                        tags=['env:{}'.format(self.environment_name),
+                              'environment:{}'.format(self.environment_name)] + self.additional_tags
+                    )
+
+                if len(interval_results) > 0:
```

Comment:
Here too, `process_topology_stats` should report its metrics. That would remove the following code.